pkg/cloud/services/resourcecontroller/service.go

```diff
@@ -43,6 +45,30 @@ func (s *Service) GetResourceInstance(getResourceInstanceOptions *resourcecontro
 	return s.client.GetResourceInstance(getResourceInstanceOptions)
 }
 
+// GetResourceInstanceDetails will get the region and zone of resource instance.
+func GetResourceInstanceDetails(serviceInstanceID string) (reg string, zone string, err error) {
```

Comment:
this function seems to me more of a util function, these functions in this package for more of a direct all to the SDKs for mocking purpose only, may be if this code is called multiple times then we can move somewhere in the util package and refer.